### PR TITLE
Feature/cs732 27

### PR DESCRIPTION
**Pull request by MarkZhuVUW**
1. 添加infra as code， 如果你们想玩AWS CDK， 需要跟我说 我给你permission deploy到我的AWS账号
2. ec2 instance在这里: http://ec2-3-107-59-53.ap-southeast-2.compute.amazonaws.com 前端是port 5000 后端是port 3000
3. 确认frontend, backend, e2eTests, infrastructure的所有package.json文件里面的scripts都能跑
4. 确认local, ci, prod三个环境都可以正常运行
5.完成所有cicd相关内容 ci有pull request的时候才会trigger, cd会在merge 进main branch的时候trigger。改动在`ci.yml` 和`cd.yml`里

32cdb7b928e65032dbbaaaeeb2a2314
4f1967199ada579a17922c6d1beed14
image

deploy aws cdk 我在cd里面暂时disable了 如果你们想deploy其他东西需要先联系我。。

### CODE REVIEW

**Comment by KevinHChen** [comment]
the `npm run start:dep` doesn't work. Please fix it.
image

Tips: it shall be `"cd ../ && sh scripts/start.sh"`

**Comment by MarkZhuVUW** [comment]
> the `npm run start:dep` doesn't work. Please fix it. image
> 
> Tips: it shall be `"cd ../ && sh scripts/start.sh"`

fixed. 
lets merge this

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -1,18 +1,114 @@
...
-  build:
+  archive-src-code:
     runs-on: ubuntu-latest
-
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up Node.js
-        uses: actions/setup-node@v1
+      - uses: actions/checkout@v4
+      # - name: Write app prod env vars
```
[the remark this answers is not in the document]
Comment (MarkZhuVUW) [reply .github/workflows/cd.yml:15]: I have another ticket to fix this lets be lazy this time

**Review by KevinHChen — APPROVED** [review]
looks good now

**Pull request merged**